File: problem003/solution.js
const biggestPrimeFactor = n => {
    let i = 2
    while (i * i <= n) {
        if (n % i) {
            i++
        } else {
            n /= i
        }
    }
    return n
}

console.log(biggestPrimeFactor(600851475143)) // 6857

File: problem003/solution.py
def biggestPrimeFactor(n):
    i = 2
    while i * i <= n:
        if n % i:
            i += 1
        else:
          # like n = int(n / 2)
            n //= i
    return n

print(biggestPrimeFactor(600851475143)) # 6857